src/coreclr/jit/morph.cpp

```diff
@@ -8135,6 +8136,34 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac, bool* optA
                         }
                     }
                 }
+
+                // Optimization for:
+                //    a & c ==/!= c
+                // =>
+                //    ~a & c ==/!= 0
+                // where c is a constant
+                // The optimization could also take the form of a & ~c but the comparison would have to switch between EQ to NE/NE to EQ
+                if (op1->OperIs(GT_AND) && varTypeIsIntegral(op1) && op2->IsIntegralConst())
+                {
+                    GenTree* andOp1 = op1->AsOp()->gtOp1;
+                    GenTree* andOp2 = op1->AsOp()->gtOp2;
+
+                    ssize_t cnsVal = op2->AsIntCon()->IconValue();
+
+                    if (andOp1->TypeIs(TYP_INT, TYP_LONG) && andOp2->IsIntegralConst() && andOp2->AsIntCon()->IconValue() == cnsVal)
+                    {
+                        // Want GT_AND to look like AND(NOT(a), c) ==/!= a. The non-matching constant must be the one wrapped in NOT node
+                        // so 2nd andOp will be the constant, so 1st andOp will have the NOT
+                        GenTree* tmpNode = andOp2;
+                        op1->AsOp()->gtOp2 = gtNewOperNode(GT_NOT, andOp1->TypeGet(), andOp1);
+                        op1->AsOp()->gtOp1 = tmpNode;
+                        op2->AsIntConCommon()->SetIconValue(0);
+#if defined(TARGET_ARM) || defined(TARGET_ARM64)
+                        // If set for XARCH it will prevent ANDN instruction from being emitted
+                        op1->gtFlags |= GTF_SET_FLAGS;
```

Comment:
Afair, GTF_SET_FLAGS is a lowering concept and shouldn't be used in morph

Reply:
Any idea how I can get bic to become bics then?

Reply:
This and the change in codegen are not quite the right way to accomplish reusing the flags. Instead you should take a look at `Lowering::TryLowerConditionToFlagsNode`, specifically the part that checks `SupportsSettingZeroFlags`, and try to update that logic to handle these cases as well.

Reply:
Thank you Jakob! This was a great hint and I figured out a generic solution.